python/ray/train/tests/test_huggingface_predictor.py

```diff
@@ -62,6 +63,52 @@ def test_repr(tmpdir):
     assert pattern.match(representation)
 
 
+def test_huggingface_checkpoint(tmpdir, ray_start_runtime_env):
+    model_config = AutoConfig.from_pretrained(model_checkpoint)
+    model = AutoModelForCausalLM.from_config(model_config)
+    tokenizer = AutoTokenizer.from_pretrained(tokenizer_checkpoint)
+    preprocessor = DummyPreprocessor()
+    preprocessor.attr = 1
+
+    checkpoint = HuggingFaceCheckpoint.from_model(
+        model, tokenizer, path=tmpdir, preprocessor=preprocessor
+    )
+    checkpoint_model = checkpoint.get_model(AutoModelForCausalLM)
+    checkpoint_tokenizer = checkpoint.get_tokenizer(AutoTokenizer)
+    checkpoint_preprocessor = checkpoint.get_preprocessor()
+
+    @ray.remote
+    def test(model, tokenizer):
+        os.chdir(tmpdir)
+        # Ensure that model outputs are deterministic
+        model_config = AutoConfig.from_pretrained(model_checkpoint)
+        model = AutoModelForCausalLM.from_config(model_config)
+        model.eval()
+        predictor = HuggingFacePredictor(
+            pipeline=pipeline(
+                task="text-generation",
+                model=model,
+                tokenizer=tokenizer,
+            ),
+            preprocessor=preprocessor,
+        )
+
+        predictions = predictor.predict(prompts)
+        return predictions
+
+    tokens = tokenizer(test_strings)
+    checkpoint_tokens = checkpoint_tokenizer(test_strings)
+
+    predictions = ray.get(test.remote(model, tokenizer))
+    checkpoint_predictions = ray.get(
+        test.remote(checkpoint_model, checkpoint_tokenizer)
+    )
+
+    assert all(predictions == checkpoint_predictions)
+    assert tokens == checkpoint_tokens
+    assert checkpoint_preprocessor.attr == preprocessor.attr
```

Comment:
Added this! There was a lot of shared `DummyPreprocessor` usage, and we were checking for a flag `_batch_transformed` manually. I moved this to a shared class that most of the predictor tests are now using, which has a `has_transformed` property instead.

So we do `assert preprocessor.has_transformed` rather than `assert hasattr(preprocessor, "_batch_transformed")`.